docs/DeveloperGuide.adoc

```diff
@@ -345,6 +345,210 @@ image::UndoRedoActivityDiagram.png[width="200"]
+
+image::SortedNumberList.png[width="800"]
+
+As the user continues to use the program, the user can request for a different sort command. For example, the user may execute sort name to interalList by their name.
+
+image::OriginalInternalList.png[width="800"]
+
+The following sequence diagram shows how the sort operation works:
+
+image::SortSequenceDiagram.png[width="800"]
+
+[NOTE]
+The user may choose to sort the persons by other parameters. The parameters includes : name, number, address, email, remark, birthday.
+
+If a command fails its execution, no sorting will be done and a parse Exception will be used to handle the format for incorrect Commands.
+
+==== Design Considerations
+
+**Aspect:** Implementation of `SortCommand` +
+**Alternative 1 (current choice):** Sort the interalList via `Collections.sort()` and  +
+**Pros:** We will be able to sort using merge or Tim sort which offers a stable and efficient (O(n log n)) efficiency +
+**Cons:** May not be optimal when sorting again after adding one new person only. +
+**Alternative 2:** Implement different algorithm depends on different type or number of persons added +
+**Pros:** Able to obtain optimal efficiency for sorting everytime. +
+**Cons:** Alot of code needed for all the different types of persons added to ensure better efficiency than Tim sort.
+
+---
+
+**Aspect:** Sorting persons based on type +
+**Alternative 1 (current choice):** Uses Lambdas to sort. +
+**Pros:** Clean one line code to compare person to sort. +
+**Cons:** - +
+**Alternative 2:** Individual Comparator for person object based on sorting parameter. +
+*Pros:** Sort the persons based on comparator, more free play to the comparisons in sorting. +
+**Cons:** Less efficient as more context switch is needed, resulting in higher overhead time.
+
+// tag::BirthdayNotifier[]
+=== Birthday and Birthday Notifier mechanism
+
+The birthday notifier mechanism is facilitated by a `BirthdayNotifier`, which resides within the `Model` component.
+It creates a pop-up every time the app is run to inform the user of any birthdays that happen to fall on that day.
+This mechanism works side by side with the `Birthday` field of each `Person`.
+
+`BirthdayNotifier` is created and run in the `MainApp` every time the app is run. `BirthdayNotifier` will then check
+through the `Birthday` of the list of `Person` that was passed in from `MainApp`. It will then create a
+`BirthdayPopup` to show the list of `Person` with birthdays on that day. These relationships are illustrated in the
+diagram below.
+
+image::BirthdayNotifierDiagram.png[width = "800"]
+
+Implementation of checking through all the `Person` in `BirthdayNotifier`:
+[source,java]
+----
+for (ReadOnlyPerson e: list) {
+    if (e.getDay() == date && e.getMonth() == month) {
+        people.add(e.getName().toString());
+    }
+}
+
+if(people.size() > 0) {
+    createPopup(people.toArray(new String[people.size()]));
+}
+----
```

Comment:
Code snippets are good, but keep them minimal as they add to your maintenance overhead. Also `if(people.size() > 0)` -> `if (people.size() > 0)`